docs/proposals/use-cascading-deletion/README.md

```diff
@@ -0,0 +1,320 @@
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the desired cascadeDeletion Behavior.
+	Spec CascadeDeletionSpec `json:"spec"`
+
+	// Status represents the status of cascadeDeletion.
+	// +optional
+	Status CascadeDeletionStatus `json:"status,omitempty"`
+}
+
+type CascadeDeletionSpec struct {
+	//CascadeDeletion Declare the cascade deletion strategy. The default value is null, which is equivalent to background.
+	// +optional
+	CascadeDeletion *CascadeDeletionPolicy `json:"cascadeDeletion,omitempty"`
+    // ResourceSelectors used to select resources.
+    // Nil or empty selector is not allowed and doesn't mean match all kinds
+    // of resources for security concerns that sensitive resources(like Secret)
+    // might be accidentally propagated.
+    // +required
+    // +kubebuilder:validation:MinItems=1
+    ResourceSelectors []ResourceSelector `json:"resourceSelectors"`
+}
+
+// ResourceSelector the resources will be selected.
+type ResourceSelector struct {
+    // APIVersion represents the API version of the target resources.
+    // +required
+    APIVersion string `json:"apiVersion"`
+    
+    // Kind represents the Kind of the target resources.
+    // +required
+    Kind string `json:"kind"`
+    
+    // Namespace of the target resource.
+    // Default is empty, which means inherit from the parent object scope.
+    // +optional
+    Namespace string `json:"namespace,omitempty"`
+    
+    // Name of the target resource.
+    // Default is empty, which means selecting all resources.
+    // +optional
+    Name string `json:"name,omitempty"`
+    
+    // A label query over a set of resources.
+    // If name is not empty, labelSelector will be ignored.
+    // +optional
+    LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+}
+
+type CascadeDeletionStatus struct {
+	...
+}
+```
+
+
+Work
+```go
+// WorkSpec defines the desired state of Work.
+type WorkSpec struct {
+    //CascadeDeletion Declare the cascade deletion strategy. The default value is null, which is equivalent to background.
+    // +optional
+    CascadeDeletion *CascadeDeletionPolicy `json:"cascadeDeletion,omitempty"`
+	// Workload represents the manifest workload to be deployed on managed cluster.
+	Workload WorkloadTemplate `json:"workload,omitempty"`
+}
+```
+#### User usage example
+
+Set the cascade deletion policy to orphan
+
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: CascadeDeletionPolicy
+metadata:
+  name: foo
+spec:
+  cascadeDeletion: orphan
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      name: foo
+      namespace: default
+```
+
+### Solution Four: Extended by Annotation & Extend the fields of PropagationPolicy/ClusterPropagationPolicy
+
+Equivalent to supporting both solution one and solution two
+
+### Solution comparison
+| Name           | Supported control plane resources | Extend API resources | User learning cost |
```

Comment:
I didn't got what is the different below the title "Supported control plane resources".